platform/core/src/classes/metadata/StudyMetadata.js

```diff
@@ -585,6 +585,38 @@ const isMultiFrame = instance => {
   return instance.getRawValue('x00280008') > 1;
 };
 
+const is4DSeries = instances => {
+  if (instances[0].getRawValue('x00080060') !== 'MR') return 0; // only MR exams
+
+  // The goal here is to identify 4D series and the tags we'll use to sort them
+  // List tags used to identify and sort 4D series
+  const sortTagsList = Object.keys(instances[0]._instance.sort4DSeriesTags);
+
+  // Among those tags, identify those that have multiple values
+  var sortTags = Array(); // Declare an array to store tags to sort by
+  var sortTagsVal = Array(); // for debug purposes
+
+  sortTagsList.map(testedTag => { // loop through the tag list
+    var testedTagValues = instances.map(b => b._instance.sort4DSeriesTags[testedTag]);
+    var tagsUniqueValues = [...new Set(testedTagValues)];
+    if (tagsUniqueValues.length > 1 && tagsUniqueValues.length < 5) {
+      // If the tested tag has more than one value throughout the series..
+      // And less than 5 (i.e. not too many)
+      sortTags.push(testedTag); // Push to our new array
+      sortTagsVal.push(tagsUniqueValues); // debug
+    }
+  });
+  // If none of them do : return 0
+  if (sortTags.length === 0) return 0;
+
+  // Debug: report on tags for each series
+  console.log('%c4D series! Sorting tags:', 'color: black; background: #9ccef9; padding: 4px; font-weight: bold;');
```

Comment:
You can use log.debug here:

https://github.com/OHIF/Viewers/blob/b59c8c277af38ab601c2f3baa5a0cef8841c4438/platform/core/src/log.js

Example usage:

https://github.com/OHIF/Viewers/blob/b59c8c277af38ab601c2f3baa5a0cef8841c4438/platform/core/src/ui/handleError.js#L1